### PR TITLE
[TEP074] Remove Storage pipelineResources

### PR DESCRIPTION
**Pull request by JeromeJu**
# Changes

Remove Storage PipelineResources

This commit removes the Storage Resources support.
This PR removes `github.com/tektoncd/pipeline/pkg/apis/resource/v1alpha1/storage`,
`pkg/artifacts` and the respective test cases.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
action required: please migrate off of `storage` `pipelineresources` as it is removed, please refer to the doc at https://github.com/tektoncd/pipeline/blob/main/docs/pipelineresources.md#replacing-an-storage-resource
`artifacts` apis are removed with the setup for pvc/buckets
```

### CODE REVIEW

File: pkg/apis/resource/v1alpha1/storage/artifact_bucket.go [hunk 1]
```diff
@@ -1,103 +0,0 @@
...
-	GsutilImage string
-}
-
-// GetType returns the type of the artifact storage
-func (b *ArtifactBucket) GetType() string {
-	return pipeline.ArtifactStorageBucketType
-}
-
-// StorageBasePath returns the path to be used to store artifacts in a pipelinerun temporary storage
-func (b *ArtifactBucket) StorageBasePath(pr *v1beta1.PipelineRun) string {
```
Comment (JeromeJu) [inline pkg/apis/resource/v1alpha1/storage/artifact_bucket.go:52]: It seems that the removal of `Storage` resources would lead to the functionalities of artifiact_bucket, it has to be handled for the pipelinerun reconciler.

**Comment by JeromeJu** [comment]
TODO:

- [ ] replace output-pipelinerun.yaml for example with resolution

**Comment by JeromeJu** [comment]
This PR is ready for review, thanks!

PTAL /assign @afrittoli @lbernick

File: docs/variables.md [hunk 2]
```diff
@@ -108,14 +108,6 @@ variable via `resources.inputs.<resourceName>.<variableName>` or
...
 
-#### Variables for the `GCS` type
-
-| Variable | Description |
-| -------- | ----------- |
-| `name` | The name of the resource. |
-| `type` | Type value of `"gcs"`. |
-| `location` | The fully qualified address of the blob storage. |
-
 #### Variables for the `Cluster` type
```
Comment (lbernick) [inline docs/variables.md:111]: it looks like there's some cloudevent and cluster docs hanging around here

File: pkg/reconciler/pipelinerun/pipelinerun.go [hunk 3]
```diff
@@ -948,7 +937,7 @@ func (c *Reconciler) createTaskRun(ctx context.Context, taskRunName string, para
 		tr.Annotations[workspace.AnnotationAffinityAssistantName] = getAffinityAssistantName(pipelinePVCWorkspaceName, pr.Name)
 	}
 
-	resources.WrapSteps(&tr.Spec, rpt.PipelineTask, rpt.ResolvedTaskResources.Inputs, rpt.ResolvedTaskResources.Outputs, storageBasePath)
+	resources.WrapSteps(&tr.Spec, rpt.PipelineTask, rpt.ResolvedTaskResources.Inputs, rpt.ResolvedTaskResources.Outputs)
```
[the remark this answers is not in the document]
Comment (lbernick) [reply pkg/reconciler/pipelinerun/pipelinerun.go:889]: @JeromeJu are you still having trouble with this bit? I think the git pipelineresource also uses artifact storage; it might be easier to not delete artifact storage in this PR or to delete the git resource and the storage resource in the same PR

File: test/custom_task_test.go [hunk 4]
```diff
@@ -1204,3 +1205,30 @@ func tearDownV1Beta1CustomTask(ctx context.Context, t *testing.T, c *clients, na
 
 	tearDown(ctx, t, c, namespace)
 }
+
+// updateConfigMap updates the config map for specified @name with values. We can't use the one from knativetest because
```
Comment (lbernick) [inline test/custom_task_test.go:1209]: was this moved from somewhere else or just a weird rebase?

File: test/dag_test.go [hunk 5]
```diff
@@ -64,13 +64,6 @@ metadata:
   name: %s
   namespace: %s
 spec:
```
Comment (lbernick) [inline test/dag_test.go:66]: seems like the changes in this file are mainly related to git resources

File: test/resolvers_test.go [hunk 6]
```diff
@@ -55,6 +55,7 @@ const (
 	scmRemoteUserPassword = "ab_d1234HIJKL"
 	// Defined in git-resolver/gitea.yaml's "gitea" StatefulSet, in the env for the "configure-gitea" init container
 	scmGiteaAdminPassword = "giteaPassword1234"
+	systemNamespace       = "tekton-pipelines"
```
Comment (lbernick) [inline test/resolvers_test.go:58]: what's the reason for this change?

Comment (JeromeJu) [reply test/resolvers_test.go:58]: It is moved from the deleted artifact_bucket_test.go

File: test/custom_task_test.go (hunk 4 above)
Comment (JeromeJu) [reply test/custom_task_test.go:1209]: It used to be in the deleted `artifact_bucekt_test.go` and referenced here.

File: pkg/reconciler/pipelinerun/pipelinerun.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (JeromeJu) [reply pkg/reconciler/pipelinerun/pipelinerun.go:889]: Thanks Lee, that makes sense, I tried to merge the changes and restore the parts on Git resources to be restored for image type at https://github.com/tektoncd/pipeline/pull/6150. 

And if that looks good I will try to make Image Resources removal as the last PR along with the removal of the generic functions as the removal of both Storage and Git resources has already been a large chunk.

[the remark this answers is not in the document]
Comment (JeromeJu) [reply pkg/reconciler/pipelinerun/pipelinerun.go:889]: Thanks @lbernick , that makes sense, I tried to merge the changes and restore the parts on Git resources to be restored for image type at https://github.com/tektoncd/pipeline/pull/6150. 

And if that looks good I will try to make Image Resources removal as the last PR along with the removal of the generic functions as the removal of both Storage and Git resources has already been a large chunk.
